### PR TITLE
Autolink helm and kubectl on startup, if possible

### PR DESCRIPTION
**Pull request by ericpromislow**
Mostly this is a refactoring of the link/unlink event handler.

### CODE REVIEW

File: background.js [hunk 1]
```diff
@@ -227,6 +216,34 @@ ipcMain.on('factory-reset', async () => {
   app.quit();
 });
 
+/**
+ * assume sync activities aren't going to be costly for a UI app.
+ * @param {String} name -- basename of the resource to link
+ * @param {Boolean} state -- true to symlink, false to delete
+ * @returns {null|Error}
```
Comment (mook-as) [inline background.js:223]:
```suggestion
 * @returns {Promise<Error?>}
```
Because this is an `async function`.

**Review by mook-as — APPROVED** [review]

File: background.js (hunk 1 above)
Comment (ericpromislow) [reply background.js]: Done. Note that the checkboxes are still broken overall in this PR -- branch `80-get-checkboxes-working` will fix them.

**Comment by mook-as** [comment]
We'll probably need a follow-up, since as-is even if the user unchecks the checkbox manually it'll be re-installed on next startup, but that's for future-us.

**Review by mattfarina — CHANGES_REQUESTED** [review]
I wanted to test what it was like to opt-out of these being in my path and ran into an issue. The kubectl I have linked into /user/local/bin is the one provided by RD. Yet, it doesn't realize it. The box is greyed out (not checked) and I see the error:

```
Reading /usr/local/bin/kubectl got error undefined result /Users/matthewfarina/Code/mattfarina/rd/resources/darwin/bin/kubectl
```

Does not look like the checkboxes are working now.

**Comment by ericpromislow** [comment]
Yup, this got rebased against 80-get-checkboxes-working so the checkboxes should be active again (if either the file is missing or symlinked to a desktop resource.

**Pull request merged**